<launch>
	<!-- global rosparam -->
	<param name="use_sim_time" value="true"/>

	<!-- sensor driver -->
	<include file="$(find velodyne_pointcloud)/launch/32e_points.launch"/>
	<node pkg="pose_estimation_walls" name="odom_republish_infant" type="odom_republish_infant"/>

	<!-- proposed method -->
	<node pkg="planar_landmark_ekf_slam" name="planar_normal_estimation" type="planar_normal_estimation">
		<param name="skip" type="int" value="3"/>
		<param name="search_radius_ratio" type="double" value="0.09"/>
		<param name="min_serach_radius" type="double" value="0.1"/>
		<!-- <param name="mode_remove_ground" type="bool" value="true"/> -->
		<!-- <param name="mode_open_viewer" type="bool" value="false"/> -->
		<!-- <param name="mode_selection" type="bool" value="false"/> -->
	</node>
	<!-- <node pkg="planar_landmark_ekf_slam" name="planar_feature_extraction" type="planar_feature_extraction"> -->
	<!-- 	<remap from="/normals" to="/selected_normals"/> -->
	<!-- 	<param name="cluster_tolerance" type="double" value="0.1"/> -->
	<!-- 	<!&#45;&#45; <param name="min_cluster_size" type="int" value="150"/> &#45;&#45;> -->
	<!-- 	<param name="min_cluster_size" type="int" value="1000"/> -->
	<!-- </node> -->
	<node pkg="planar_landmark_ekf_slam" name="planar_feature_extraction_flexible_tolerance" type="planar_feature_extraction_flexible_tolerance" output="screen">
		<remap from="/normals" to="/normals/selected"/>
		<param name="ratio_depth_tolerance" type="double" value="0.05"/>
		<param name="min_tolerance" type="double" value="0.1"/>
		<param name="max_tolerance" type="double" value="0.3"/>
		<param name="min_cluster_size" type="int" value="500"/>
		<param name="threshold_angle" type="double" value="1.0"/>
	</node>
</launch>
